### ISSUE TITLE
Processing 14-month runs (PAMIP)

### ISSUE THREAD

**Issue by ypeings** [E3SM-Project/e3sm_to_cmip#65]
Dear E3SM colleagues,

In our team at UC Irvine, we have run E3SMv1 simulations for the Polar Amplification MIP (PAMIP). These experiments consists of ensembles of 14-month runs, going from April of year 0 to May of year +1.

I am trying to use e3sm_to_cmip to CMOR this data and make it available on ESGF. I see in the instructions that the input data has to be processed using ncclimo, to have one file per variable from a start, to an end year date (VAR_STARTYEAR_ENDYEAR.nc). In our case each file would correspond to a 14-month timeserie, but it seems ncclimo wants full years of data. I get the following error when trying to use ncclimo : 
ncclimo: HINT Provide input filenames in multiples of 12

Is there a workaround for this ? How can I process our data to feed e3sm_to_cmip ?
Any help is much appreciated. Thanks !

**Comment by chengzhuzhang** [comment]
Hi Charlie @czender do we have a possible workaround for splitting time-series from 14-month data.

**Comment by czender** [comment]
What is the context? Is `ncclimo` being called in splitter mode, i.e., to separate time series of multi-variable files into timeseries of single-variable files? Or what? An example of an `ncclimo` command and a listing of the (14?) readable input files would be helpful.

**Comment by ypeings** [comment]
@czender My files are not timeseries, they only include one month (h0 files) or day (h1 files). There is an example here : 
/global/cscratch1/sd/ypeings/e3sm_scratch/cori-knl/archive/PAMIP-1.6_20/atm/hist

ls *h0* will show you the 14 monthly files that include all the variables. My goal is to process these files so I can use e3sm_to_cmip to convert them to CMIP format. I hav enever used ncclimo before, but the e3sm_to_cmip instructions state the input files has to be formatted by ncclimo. 
Thanks for your help

**Comment by czender** [comment]
Double-checking that the files you want to produce in CMIP format are 14-month long timeseries, one timeseries per variable? Or what?

**Comment by ypeings** [comment]
@czender Yes, that's what we want to produce, one 14-month long file per variable.

**Comment by czender** [comment]
Hi Yannick,
I looked into this and will create a patch for `ncclimo` that does this. It is a gap in functionality that I have known would need to be fixed for a long time. The patch will simply require the user to add arguments for the start and end month. Nothing too painful. However, the internal changes necessary are non-trivial and, as importantly, I am slammed for the next two weeks. I think I will have this done about three weeks from now. I'll post again when I have a version you can test.
Charlie

**Comment by ypeings** [comment]
@czender This is awesome, thank you very much Charlie !

**Comment by czender** [comment]
FYI this is almost finished. Final changes will be done within a few days.

**Comment by czender** [comment]
OK, it's ready. Try adding mth_srt and mth_end to your syntax like:

```
drc_in=${DATA}/ne30/raw
drc_out=${DATA}/ne30/clm
caseid='20180129.DECKv1b_piControl.ne30_oEC.edison.cam.h0.'
cd ${drc_in};ncclimo --dbg=1 -v FSNS --yr_srt=1 --yr_end=2 --mth_srt=4 --mth_end=5 --drc_out=${drc_out} ${caseid}0001-0{4..9}.nc ${caseid}0001-1{0..2}.nc ${caseid}0002-0{1..5}.nc
```
Just be sure to use ncclimo v. 4.9.8-alpha03 or later. You can copy it or run it from `~zender/bin_cori/ncclimo` on Cori. LMK how it works...

**Comment by ypeings** [comment]
@czender  Hi Charlie, thank you so much ! I tried it, it works perfectly, this will be very helpful for sharing our PAMIP data on ESGF.
Thanks again and have a great day !

**Comment by czender** [comment]
Glad to hear it, Yannick. This will also be released in NCO 4.9.8 within a few weeks.
